src/DeFuncto.Tests/Core/Tests/Try.cs

```diff
@@ -0,0 +1,39 @@
+﻿using FsCheck.Xunit;
+using DeFuncto.Assertions;
+using System;
+using FsCheck;
+using System.Threading.Tasks;
+
+namespace DeFuncto.Tests.Core.Tries
+{
+    public class Try
+    {
+        [Property]
+        public void TryOk(int a) =>
+            DeFuncto.Extensions.Tests.Try(() => a)
+                .ShouldBeOk(a);
+
+        [Property]
+        public void TryError(NonNull<string> a)
+        {
+            Exception expected = new Exception(a.Get);
+
+            DeFuncto.Extensions.Tests.Try(() => { throw expected; return a; })
+                .ShouldBeError(expected);
+        }
+
+        [Property]
+        public void TryAsyncOk(int a) =>
+            DeFuncto.Extensions.Tests.Try(() => Task.FromResult(a))
+                .ShouldBeOk(a);
+
+        [Property]
+        public void TryAsyncError(NonNull<string> a)
+        {
+            Exception expected = new Exception(a.Get);
+
+            DeFuncto.Extensions.Tests.Try(async () => { throw expected; return a; })
```

Comment:
You are not awaiting, this might be not working.
Doing TDD this would have popped out.